opencti-platform/opencti-graphql/src/migrations/1712225747215-convert_linked-to_related-to.js

```diff
@@ -0,0 +1,128 @@
+import { logApp } from '../config/conf';
+import { elDeleteByQueryForMigration, elList, elReindexByQueryForMigration, elUpdateByQueryForMigration } from '../database/engine';
+import { INDEX_STIX_CORE_RELATIONSHIPS, READ_DATA_INDICES, INDEX_STIX_META_RELATIONSHIPS, READ_INDEX_STIX_META_RELATIONSHIPS, READ_RELATIONSHIPS_INDICES } from '../database/utils';
+import { RELATION_RELATED_TO } from '../schema/stixCoreRelationship';
+import { executionContext, SYSTEM_USER } from '../utils/access';
+
+export const up = async (next) => {
+  const context = executionContext('migration');
+  logApp.info('[MIGRATION] Update all linked-to refs to related-to rels');
+
+  const linkedToType = 'x_opencti_linked-to';
+  const relLinkedTo = `rel_${linkedToType}.internal_id`;
+  const relRelatedTo = `rel_${RELATION_RELATED_TO}.internal_id`;
+
+  // We get all current related-to relations: will be used to check for duplicates in linked-to to related-to migration
+  const relatedToRelations = await elList(context, SYSTEM_USER, READ_RELATIONSHIPS_INDICES, { types: [RELATION_RELATED_TO] });
```

Comment:
This might list a lot of relationships, no ? compared to linked-to refs, I wonder if we couldn't list linked-to relationships first then search for duplicates in related_to, and exclude these duplicates ? And if we don't find any linked-to, then we don't even have to continue the migration.

Reply:
you're right, I didn't keep in mind that it was only the first pagination. I'll change it to what you suggested